styled.d.ts

```diff
@@ -0,0 +1,59 @@
+import "styled-components";
+
+declare module "styled-components" {
+    export type StandardPaletteList = {
+        background: string[];
+        foreground: string[];
+        largeForeground: string[];
+    };
+
+    export type StandardPalette = {
+        background: string;
+        foreground: string;
+        largeForeground: string;
+    };
+
+    export type RainbowPalette = {
+        lightRed: StandardPalette;
+        darkRed: StandardPalette;
+        lightOrange: StandardPalette;
+        darkOrange: StandardPalette;
+        lightYellow: StandardPalette;
+        darkYellow: StandardPalette;
+        lightGreen: StandardPalette;
+        darkGreen: StandardPalette;
+        lightBlue: StandardPalette;
+        darkBlue: StandardPalette;
+        lightPurple: StandardPalette;
+        darkPurple: StandardPalette;
+        lightGrey: StandardPalette;
+        darkGrey: StandardPalette;
+        lightBrown: StandardPalette;
+        darkBrown: StandardPalette;
+    };
+
+    /**
```

Comment:
this is quite cool! is this a docstring that is rendered somewhere in IDEs?

Reply:
Yepp! So if you hover to the word `theme` in `props.theme.xxx` and click the `DefaultTheme` it will teach you how to use it
Just want to make others life easier when trying to replicate pages to use similar format so they don't have to spend too much time in understanding how other pages css their colors